### PR TITLE
Add content for BCI support in class data sharing

### PR DESCRIPTION
**Pull request by SueChaplain**
- new section on BCI support and sharing modified bytecode in the cache.
- a few minor updates to other shared classes sections.

Signed-off-by: SueChaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

File: docs/shrc.md [hunk 1]
```diff
@@ -285,6 +288,39 @@ The following APIs are available for obtaining information about a shared classe
...
+
+By default, if OpenJ9 detects that a JVMTI agent or `java.lang.instrument` agent has registered to modify class bytes, modified classes are not stored in the cache. Instead, the VM stores original class byte data in the cache, which allows classes to be retransformed.
+
+If you turn off bytecode instrumentation support by specifying [`-Xshareclasses:disableBCI`](xshareclasses.md#disablebci), all bytecode is loaded from the file system for the agent to modify. When passed to the cache for storing, the VM compares the bytes with known classes of the same name as a safety measure. If a match is found, the class is reused. However, if a match is not found, the potentially modified class is stored in the cache in a way that prevents other VMs from loading it. In this situation, performance can be affected because the bytecode is always loaded from the file system. When bytecode instrumentation is turned off, classes cannot be retransformed.
```
Comment (hangshao0) [inline docs/shrc.md:297]: > If you turn off bytecode instrumentation support by specifying -Xshareclasses:disableBCI, all bytecode is loaded from the file system for the agent to modify.

This is true if you disable BCI support and do not use modification context (by default, there is no modification context). Not sure if it is easy for all user to infer this from the later paragraph talking about modification context.

File: docs/shrc.md [hunk 2]
```diff
@@ -236,6 +236,7 @@ Caches can be deleted if they contain many stale classes or if the cache is full
 - [`-Xshareclasses:destroyAllSnapshots`](xshareclasses.md#destroyallsnapshots): Removes all cache snapshots from disk that are found by specifying the `cacheDir` suboption.
 - [`-Xshareclasses:destroyAllLayers`](xshareclasses.md#destroyalllayers): Removes all shared cache layers that are specified by the `name` and `cacheDir` suboptions.
 
+:fontawesome-solid-pencil-alt:{: .note aria-hidden="true"} **Note:** Although you can remove persistent caches and snapshot files from the file system directly, you must always use the utilities to remove non-persistent caches correctly from shared memory.
```
Comment (hangshao0) [inline docs/shrc.md:239]: > Although you can remove persistent caches and snapshot files from the file system directly

Yes, they can do that. But we may not want to mention this in the doc. We always encourage users to use the utilities to remove caches. Probably remove this and just say

"You must always use the utilities to remove non-persistent caches correctly from shared memory."

File: docs/shrc.md (hunk 1 above)
Comment (SueChaplain) [reply docs/shrc.md]: Thanks - I've updated this paragraph and linked to the sharing section.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Review by hangshao0 — APPROVED** [review]

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
@pshipton - changes ready.

**Review by pshipton — APPROVED** [review]

**Pull request merged**